### PR TITLE
replace EDM calcuation with FFV form factors

### PR DESCRIPTION
**Pull request by wkotlarski**
The relative difference compared to Jobst's calculation for all leptons is of the order 1e-14 (provided we don't discard SM contribution). If we do, the differences are:
- e: 1.3156e-10
- mu: 2.97626e-06
- tau: 0.000444331

### CODE REVIEW

**Comment by Expander** [comment]
Regarding discarding the SM contribution: I have the feeling that it would be better to discard the SM contribution, because it would allow users to take our EDM contribution and add it to known (multi-loop) SM contributions. What do you think?

**Comment by wkotlarski** [comment]
You're probably right but I wouldn't touch it right now. Once the new code reproduces what we had before (modulo the change from new diagrams from Douglas) we should merge it. And then discuss it while we prepare 3.0 manual.

**Comment by wkotlarski** [comment]
This PR is ready to be merged. In `CMSSMCPV` it reproduces exactly results from Jobst's calculation (if we disable vector diagrams from Douglas).

File: templates/edm.cpp.in [hunk 1]
```diff
@@ -24,213 +24,40 @@
...
+@EDMCalculation@
 
-   return 1.0 / (2.0 * d * d) * (3.0 - r - 2.0 * std::log(r) / d);
-}
-
-double OneLoopFunctionB(double r)
-{
-   if (is_zero(r))
-      return 1.0/2.0;
+   double val =
```
Comment (Expander) [inline templates/edm.cpp.in:54]: I'd recommend to make this variable `const`.

File: templates/edm.cpp.in [hunk 2]
```diff
@@ -24,213 +24,40 @@
...
-
-      const double massRatioSquared = Sqr(exchangeFieldMass / photonEmitterMass);
-      const double loopFactor = photonEmitterCharge * OneLoopFunctionB(massRatioSquared);
-      const double contribution = numericFactor * massFactor * couplingFactor * loopFactor;
-
-      res += contribution;
-   }
-
-   return res;
 }
```
Comment (Expander) [inline templates/edm.cpp.in:62]: Please add
~~~.cpp
// namespace CMSSMCPV_edm
~~~
as comment here

**Comment by Expander** [comment]
I really like this PR! Please feel free to merge it. (You may consider making the three small suggested changes.)

**Review by Expander — APPROVED** [review]

File: templates/edm.cpp.in [hunk 3]
```diff
@@ -24,213 +24,40 @@
...
-#define MODELPARAMETER(p) context.model.get_##p()
-#define DERIVEDPARAMETER(p) context.model.p()
-#define PHASE(p) context.model.get_##p()
+#include "@ModelName@_FFV_form_factors.hpp"
 
 using namespace flexiblesusy;
 using namespace @ModelName@_cxx_diagrams;
 
 namespace {
 static constexpr double oneOver16PiSquared = 0.0063325739776461107152;
```
[the remark this answers is not in the document]
Comment (wkotlarski) [reply templates/edm.cpp.in]: Done in c1cdb1eb0e1c27f1b5ac143bac216e93cec6bc71.

File: templates/edm.cpp.in (hunk 1 above)
Comment (wkotlarski) [reply templates/edm.cpp.in]: Done in 18b091268d2ffc1ad4a97c97771326d536681ea4.

File: templates/edm.cpp.in (hunk 2 above)
Comment (wkotlarski) [reply templates/edm.cpp.in]: Done in 80f97a15840a171faf0e764a0ebdfe6da432a8bc.

**Pull request merged**